pkg/controllers/provisioningv2/cluster/controller.go

```diff
@@ -308,9 +308,49 @@ func mgmtClusterName() (string, error) {
 	return name.SafeConcatName("c", "m", rand[:8]), nil
 }
 
+func (h *handler) updateImportedCluster(cluster *v1.Cluster, status v1.ClusterStatus, mgmtCluster *v3.Cluster) ([]runtime.Object, v1.ClusterStatus, error) {
+	newCluster := &v3.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        mgmtCluster.Name,
+			Labels:      mgmtCluster.Labels,
+			Annotations: mgmtCluster.Annotations,
+		},
+		Spec: mgmtCluster.Spec,
+	}
+
+	if features.ProvisioningV2FleetWorkspaceBackPopulation.Enabled() {
```

Comment:
Does this let you change the fleet workspace name after the provisioning cluster is already created? If so what happens, does the old provisioning cluster get orphaned?